### PR TITLE
Map various render buffer / buffer builder internals

### PR DESCRIPTION
**Pull request by Sollace**
I'm sure some of this was already mapped with my PR for RenderPhase, but I guess they were corrupted by the move to 21w14a.

### CODE REVIEW

**Comment by liach** [comment]
@Sollace Might need you to look at #2312 stuff and decide which ones are better

**Comment by Sollace** [comment]
@liach I've changed this PR to a draft whilst I look through the differences. I may take some things from your PR where it looks like you got it more correct. Is that okay?

**Comment by liach** [comment]
Sure, you do devise better names than I do!

File: mappings/net/minecraft/client/render/VertexFormat.mapping [hunk 1]
```diff
@@ -10,12 +16,16 @@ CLASS net/minecraft/class_293 net/minecraft/client/render/VertexFormat
 	METHOD method_22649 startDrawing ()V
 	METHOD method_22651 endDrawing ()V
 	METHOD method_34445 getShaderAttributes ()Lcom/google/common/collect/ImmutableList;
+	METHOD method_34446 getVertexArray ()I
+	METHOD method_34447 getVertexBufferId ()I
```
Comment (liach) [inline mappings/net/minecraft/client/render/VertexFormat.mapping]:
```suggestion
	METHOD method_34447 getVertexBuffer ()I
```
to be consistent with other id getters.

File: mappings/net/minecraft/client/render/VertexFormatElement.mapping [hunk 2]
```diff
@@ -1,54 +1,135 @@
...
+		METHOD method_22667 (IIIJII)V
+			ARG 0 size
+			ARG 1 type
+			ARG 2 stride
+			ARG 3 pointer
+			ARG 5 textureIndex
 			ARG 6 elementIndex
+		METHOD method_34452 (II)V
+			ARG 0 texture_index
+			ARG 1 element_index
```
Comment (liach) [inline mappings/net/minecraft/client/render/VertexFormatElement.mapping]: should use lower camel than snake case

File: mappings/net/minecraft/client/render/entity/FishingBobberEntityRenderer.mapping [hunk 3]
```diff
@@ -1,3 +1,21 @@
 CLASS net/minecraft/class_906 net/minecraft/client/render/entity/FishingBobberEntityRenderer
 	FIELD field_21742 LAYER Lnet/minecraft/class_1921;
 	FIELD field_4707 TEXTURE Lnet/minecraft/class_2960;
+	METHOD method_23172 (FFFLnet/minecraft/class_4588;Lnet/minecraft/class_4587$class_4665;FF)V
+		ARG 0 x
+		ARG 1 y
+		ARG 2 z
+		ARG 3 buffer
+		ARG 4 mormal
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/FishingBobberEntityRenderer.mapping]:
```suggestion
		ARG 4 normal
```

**Comment by liach** [comment]
this looks great!

File: mappings/net/minecraft/client/render/VertexFormatElement.mapping (hunk 2 above)
Comment (Sollace) [reply mappings/net/minecraft/client/render/VertexFormatElement.mapping]: Whoops. How did that happen? 🙄

Comment (Sollace) [reply mappings/net/minecraft/client/render/VertexFormatElement.mapping]:
```suggestion
			ARG 1 elementIndex
```

File: mappings/net/minecraft/client/render/VertexFormatElement.mapping [hunk 4]
```diff
@@ -1,54 +1,135 @@
...
+			ARG 6 elementIndex
+		METHOD method_22667 (IIIJII)V
+			ARG 0 size
+			ARG 1 type
+			ARG 2 stride
+			ARG 3 pointer
+			ARG 5 textureIndex
 			ARG 6 elementIndex
+		METHOD method_34452 (II)V
+			ARG 0 texture_index
```
Comment (Sollace) [inline mappings/net/minecraft/client/render/VertexFormatElement.mapping]:
```suggestion
			ARG 0 textureIndex
```

**Comment by liach** [comment]
I am a bit busy right now. Will update my pr to remove the duplicate buffer stuff (I mapped a bit world renderer stuff not covered here)

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Pull request merged**